### PR TITLE
Test coverage for dispatchables in loans pallet #31

### PR DESCRIPTION
**Pull request by yz89**
1. Add essential unit test for loans pallet #31
2. Add some rust document for loans pallet #32

### CODE REVIEW

File: pallets/loans/src/mock.rs [hunk 1]
```diff
@@ -129,17 +132,37 @@ impl pallet_balances::Config for Runtime {
     type WeightInfo = pallet_balances::weights::SubstrateWeight<Runtime>;
 }
 
+lazy_static! {
+    pub static ref PRICE_ORACLE: Mutex<HashMap<CurrencyId, Option<PriceDetail>>> = {
+        Mutex::new(
+            vec![DOT, KSM, BTC, USDT, XDOT]
```
Comment (kaichaosun) [inline pallets/loans/src/mock.rs:138]: BTC will not be supported until further discussion, we should remove it from code and test cases.

File: pallets/loans/src/mock.rs [hunk 2]
```diff
@@ -129,17 +132,37 @@ impl pallet_balances::Config for Runtime {
...
+                .iter()
+                .map(|&x| (x, Some((1, 1))))
+                .collect(),
+        )
+    };
+}
+
+impl PRICE_ORACLE {
```
Comment (kaichaosun) [inline pallets/loans/src/mock.rs:146]: How about `MockPriceFeeder`?

File: primitives/src/lib.rs [hunk 3]
```diff
@@ -87,7 +87,7 @@ pub type BlockId = generic::BlockId<Block>;
 pub use sp_runtime::OpaqueExtrinsic as UncheckedExtrinsic;
 
 #[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug, PartialOrd, Ord)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash))]
```
Comment (kaichaosun) [inline primitives/src/lib.rs:90]: Can you explain about the change here?

**Review by kaichaosun — APPROVED** [review]
Looks good 👍

File: pallets/loans/src/mock.rs (hunk 2 above)
Comment (yz89) [reply pallets/loans/src/mock.rs:146]: Good idea, I will do that.

File: primitives/src/lib.rs (hunk 3 above)
Comment (yz89) [reply primitives/src/lib.rs:90]: The trait `std::hash::Hash` should be satisfied by CurrencyId enum when use it as the key of  `std::collections::HashMap`.

File: pallets/loans/src/mock.rs (hunk 1 above)
Comment (yz89) [reply pallets/loans/src/mock.rs:138]: All the currency types in the lending market will be listed by the governance module. Current markets are more of a demonstration.

File: pallets/loans/src/mock.rs (hunk 2 above)
Comment (yz89) [reply pallets/loans/src/mock.rs:146]: I will rename it to `MOCK_PRICE_FEEDER` because the static variable should have an upper case name.

**Review by MrPai — APPROVED** [review]
looks good!

**Pull request merged**